src/parser/priest/discipline/modules/azeritetraits/DeathThroes.js

```diff
@@ -54,66 +52,84 @@ class DeathThroes extends Analyzer {
-    // Reduce damage based on the disc priest damage nerf.
+    const { damage } = deathThroesStats(
+      this.selectedCombatant.traitsBySpellId[SPELLS.DEATH_THROES.id],
+    );
+
     this.damageValue = damage * (1 - DISC_PRIEST_DAMAGE_REDUCTION);
+    this.addEventListener(
+      this.atonementAnalyzer.atonementEventFilter.by(SELECTED_PLAYER),
+      this.atonementHandler,
+    );
+    this.addEventListener(
+      this.atonementAnalyzer.atonementDamageSourceFilter,
+      this.handleDeathThroesDamage,
+    );
   }
 
-  on_byPlayer_damage(event) {
-    const spellId = event.ability.guid;
-    if (spellId !== SPELLS.SHADOW_WORD_PAIN.id && spellId !== SPELLS.PURGE_THE_WICKED_BUFF.id && !event.ability.tick) {
-      return;
-    }
-    this.damageDone += this.damageValue;
+  /**
+   * Determines if a damage event is suitable to be parsed by this module
+   * 
+   * @param {Object} damageEvent A damage event
+   */
+  static validTriggerEvent(damageEvent) {
+    const spellId = damageEvent.ability.guid;
+    return ((spellId === SPELLS.SHADOW_WORD_PAIN.id || spellId === SPELLS.PURGE_THE_WICKED_BUFF.id) && damageEvent.tick);
   }
 
-  on_byPlayer_heal(healingEvent) {
-    const damageEvent = this.atonementDamageSource.event;
-
-    // Skip any healing that isn't from an atonement event
-    if (!isAtonement(healingEvent) || !damageEvent) {
-      return;
-    }
-
-    // Skip any healing that isn't the result of a SWP cast or a PtW cast
-    if (damageEvent.ability.guid !== SPELLS.SHADOW_WORD_PAIN.id && damageEvent.ability.guid !== SPELLS.PURGE_THE_WICKED_BUFF.id) {
-      return;
-    }
-
-    // Skip any healing that isn't from a tick. Death Throes only increases damage from ticks.
-    if (!damageEvent.tick) {
-      return;
-    }
-
-    if (damageEvent.amount === 0) {
-      return;
-    }
-
-    // Calculate how much of the healing should be attributed to Death Throes
-    // The damage can't be more than the damage value for DT, nor can it be more than the damage that this ability did.
-    // This prevents us from saying DT did more than 100% damage
-    let realDamage;
-    if (damageEvent.hitType === HIT_TYPES.CRIT) {
-      realDamage = Math.min(this.damageValue * 2, damageEvent.amount);
-    } else {
-      realDamage = Math.min(this.damageValue, damageEvent.amount);
-    }
-    const deathThroesDamagePercent = realDamage / damageEvent.amount;
+  /**
+   * Processing this here ensures that the damage event is only considered once,
+   * so we don't get super inflated values for damage contribution
+   *
+   * @param {Object} damageEvent A damage event
+   */
+  handleDeathThroesDamage(damageEvent) {
+    if (!DeathThroes.validTriggerEvent(damageEvent)) return;
+    const totalDamageAmount = damageEvent.amount + damageEvent.absorbed;
+    const damageRatio = totalDamageAmount / damageEvent.unmitigatedAmount;
+    const dtContribution = this.damageValue * damageRatio;
```

Comment:
need to check for crit and scale with versa here

Reply:
I pm'ed you this, but I'll put it here too. `unmitigatedAmount` doesn't change if an ability critical strikes, so even if it is a crit it will show the base amount of the spell before a crit. This means we don't need to explicitly handle crit here, versatility on the other hand does need to be.